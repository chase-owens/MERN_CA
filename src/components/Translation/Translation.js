import React from 'react';
import { theme } from '../../styles/theme';
import { isMobile } from 'react-device-detect';
import { withStyles } from '@material-ui/core/styles';
import withWidth from '@material-ui/core/withWidth';
import { unstable_useMediaQuery as useMediaQuery } from '@material-ui/core/useMediaQuery';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Chip from '../Chip/Chip';
import { Link } from 'react-router-dom';

const styles = theme => ({
  chip: {
    '& span': {
      fontSize: isMobile && 5
    }
  }
});

const Translation = ({ classes }) => {
  const color = useMediaQuery(theme.breakpoints.down(740))
    ? 'textSecondary'
    : 'default';
  return (
    <div>
      <Typography color={color} paragraph variant='subtitle2'>
        Translation
      </Typography>
      <Grid container spacing={8} direction='row' wrap='wrap'>
        <Grid item>
          <Chip className={classes.chip} value='gen' label='General' />
        </Grid>
        <Grid item>
          <Chip className={classes.chip} value='ba' label='Behavior Analysts' />
        </Grid>
        <Grid item>
          <Chip className={classes.chip} value='robo' label='Canine' />
        </Grid>
      </Grid>
    </div>
  );
};

export default withStyles(styles)(withWidth({ withTheme: true })(Translation));
