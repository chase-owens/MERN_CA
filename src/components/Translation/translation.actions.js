export const changeAudience = audience => ({
  type: CHANGE_AUDIENCE,
  audience
});

export const CHANGE_AUDIENCE = 'CHANGE_AUDIENCE';
