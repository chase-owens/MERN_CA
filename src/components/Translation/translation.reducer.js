import { defaultState } from '../../server/defaultState.js';
import { CHANGE_AUDIENCE } from './translation.actions';

export const audienceReducer = (state = defaultState.audience, action) => {
  switch (action.type) {
    case CHANGE_AUDIENCE:
      return { audience: action.audience };
    default:
      return state;
  }
};
