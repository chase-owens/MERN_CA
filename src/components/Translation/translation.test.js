import React from 'react';
import Translation from './Translation';
import { shallow } from 'enzyme';

describe('Translation', () => {
  it('renders without crashing', () => {
    const translation = shallow(<Translation />);
    expect(translation).toMatchSnapshot();
  });
});
