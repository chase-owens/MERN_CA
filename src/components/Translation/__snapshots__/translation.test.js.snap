// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translation renders without crashing 1`] = `
<WithTheme(WithWidth(Translation))
  classes={
    Object {
      "chip": "WithTheme-WithWidth-Translation---chip-1",
    }
  }
/>
`;
